README.md

```diff
@@ -1,116 +1,165 @@
+
+**Code Overview:**
+
+- **`RetNetModelHF`**: Implements the RetNet architecture as a subclass of PreTrainedModel, using Hugging Face's utilities and standards for model configuration, serialization, and compatibility with the Transformers library.
+- **`TransformerModelHF`**: Implements the Transformer architecture as a subclass of PreTrainedModel, using Hugging Face's utilities and standards for model configuration, serialization, and compatibility with the Transformers library.
+- **Configuration Classes**: Both models utilize specific configuration classes (`RetNetConfig` for RetNetModelHF and `DecoderConfig` for TransformerModelHF) to define model parameters, ensuring flexibility and ease of customization.
+
+**Usage:**
+
+To use these models within your Hugging Face-based projects, follow these steps:
+
+1. **Initialization**: Instantiate the model with the desired configuration, which can be a predefined object, a path to a configuration file, or left as default for automatic configuration.
+
+   ```python
+   from <YOUR_MODULE> import RetNetModelHF, TransformerModelHF
+
+   retnet_model = RetNetModelHF(config="path/to/retnet/config")
+   transformer_model = TransformerModelHF(config="path/to/transformer/config")
+   ```
+
+2. **Forward Pass**: Call the model with input data tensors to receive output predictions.
+
+   ```python
+   input_ids = ...  # Your input tensor here
+   retnet_output = retnet_model(input_ids)
+   transformer_output = transformer_model(input_ids)
+   ```
+
+3. **Parameter Access**: Retrieve model hyperparameters for inspection or further processing.
+
+   ```python
+   retnet_params = retnet_model.get_params()
+   transformer_params = transformer_model.get_params()
+   ```
+
+## Benchmarking and Results
+
+We use EleutherAI's open-source language model evaluation harness to empirically evaluate our models across a suite of different NLP tasks. Run the evaluation suite as follows:
+First, edit the 'tasks' parameter in the YAML file. Specify all tasks you would like to run, e.g.,
+```
+tasks:
+  - "hellaswag"
+  - "winogrande"
+```
+Alternatively, you can use `tasks: '*'` to run all benchmarks in the suite. Then navigate to the `slurm/run_eval.sh`, copy the script, and subsitute your yaml file for the placeholder. Finally, execute: 
 
-```yaml
-learning_rate: 0.001
-batch_size: 64
-num_layers: 4
-embed_dim: 256
 ```
+mamba activate <YOUR_ENV_HERE> # Activate environment, if using one.
+cd /301r_retnet/slurm/
+cp run_eval.sh user_slurm/<NAME_OF_NEW_FILE>.sh  # Give your file a descriptive name, (e.g., 'retnet_40540_run_eval.sh')
```

Comment:
Why do we all the sudden switch convention to making comments in-line instead of directly above the line they're referencing?

Reply:
Formatting issue is resolved.